app/components/candidate_interface/previous_applications_component.rb

```diff
@@ -0,0 +1,70 @@
+module CandidateInterface
+  class PreviousApplicationsComponent < ViewComponent::Base
+    include ViewHelper
+
+    def initialize(candidate:)
+      @candidate = candidate
+    end
+
+    def render?
+      application_choices.present?
+    end
+
+    def application_choices
+      (previous_application_choices + eligible_current_application_choices)
+        .compact
+        .sort_by(&:id)
+    end
+
+    def provider_name_for(application_choice)
+      application_choice
+        .course
+        .provider
+        .name
+    end
```

Comment:
```
application_choice
        .provider
        .name
```

will work here as it has through relationship with course 

![image](https://user-images.githubusercontent.com/42515961/112834806-5a52d380-9090-11eb-94f2-29e6408bda35.png)